### DISCUSSION TITLE
Vtiger

### DISCUSSION THREAD

**Discussion by bpabiszczak** [YetiForceCompany/YetiForceCRM#16314] [General] [answered]
Since we received some questions via email and the topic was also discussed on the Vtiger forum, we decided to open our own discussion for everyone interested in possibly migrating their business from Vtiger to YetiForce. 

For this purpose we collected some of the questions asked on the forum or sent directly to us in order to answer them and begin substantial discussion. 

1. The first issue you mentioned were some really good addons from Its4You (EmailMaker & PDFMaker)
Unfortunately nobody described exactly what’s missing in our built-in mechanisms by default and I can’t expect Its4You to do that since there’s nothing in it for them… I assume these addons contribute greatly to their income, meanwhile most of these functionalities are free and built-in in YetiForce. 

Someone would have to make a comparison. We don’t have time for it, and even if we did - we’d be accused of not being objective. Their solutions are definitely more intuitive but I’m not sure how much more functional they are. 

2. Workflow Designer from Redoo networks
Similarly to the first question, I assume users who don’t know YetiForce well don’t know what automation mechanisms are built into the system. This extension has a great interface but I doubt it offers even part of the possibilities YetiForce offers by default. 

Maybe someone could make a comparison, but IMO it’s fear of the unknown :)

3. System registration issue
We understand the best case scenario would be no registration at all… it’s not an indisputable necessity for us. We could possibly let go of the registration requirement, but it’s great to know some basic and publicly available information about the companies that use our system. 

If this is a deal breaker for some of the people to switch to YetiForce then we could change it. The question is, however, how many people consider it a deal breaker? We don’t want to end up doing this for just one person/company.

4. What limitations apply to the “Community” version?
There is just one single version of YetiForce. There are no limitations. You can create addons, plugins, and components. 95% of our clients use the latest version. What could be considered a disadvantage is the fact that all our addons (including the paid ones) are built into the system and they’re available even if someone doesn’t need them. 

5. Questions about reports, mobile apps, charts…
YetiForce has all these mechanisms, they might work differently than they do in Vtiger, but in my opinion they are better. In order to evaluate the system you have to get familiar with it  - otherwise it will be difficult to compare Vtiger and YetiForce since there have been over 40.000 changes! These systems are different in every aspect. 

**To sum up**

You need to get to know YetiForce not only from the perspective of what you think it can do, but from the perspective of what has been added over the last 8 years. The number of additional mechanisms provides you and your clients with great possibilities.

If there’s something you can’t find - let us know. We’ll send you an article that answers your questions.

**Comment by rulotec** [comment]
Regarding 1. One thing I did notice is that in yetiforce templates you cannot have custom functions. For example, lets say I wan't to display a number with words. Eg. Display "Thrity two" instead of 32. With a function `number2words` you could achieve that. One more thing is that pdfmaker/email maker allows you to introduce if statements in the templates. Maybe I just didn't find the functionality in Yeti, so please correct me if I'm wrong.

3. According to the Yeti license "All provided registration information, including the company’s name, can be used by YetiForce S.A. (Ltd.) for marketing purposes". In my particular case I would like to provide hosting services. What preoccupies me is the possibility of you contacting my customers directly and getting them to switch to Yeti hosting solutions.

4. I'd like to create plugins for my customers. Is there a way to install them via a zip file?

5. I searched and couldn't find a reports module. Could you tell me where to find reports?

**Reply by rulotec** [reply]
I wouldn't like my clients to see this:

![image](https://user-images.githubusercontent.com/36896420/182420405-c47345d9-2232-410e-80a2-157d1cf88aeb.png)

and this:

![image](https://user-images.githubusercontent.com/36896420/182420501-3b4ef31b-2692-423d-a245-98ce0a816c26.png)

**Reply by apcloic** [reply]
@rulotec 
Yes it looks like you can install module from file.
![image](https://user-images.githubusercontent.com/2930560/182646817-a7df94f0-dec4-45e1-90af-be42435e8755.png)

**Reply by rulotec** [reply]
Thanks @apcloic I didn't see that option in the demo. But I guess it's disabled for security reasons.

**Reply by mariuszkrzaczkowski** [reply]
> One more thing is that pdfmaker/email maker allows you to introduce if statements in the templates. Maybe I just didn't find the functionality in Yeti, so please correct me if I'm wrong.

The Twig engine was used in the PDF templates, so you can safely use loop conditions etc.
https://github.com/YetiForceCompany/YetiForceCRM/blob/33653abe468c2f1734a9dd6c4c5c2b920541e1a5/app/Extension/Twig/SecurityPolicy.php#L21-L23
https://twig.symfony.com/doc/3.x/coding_standards.html
https://twig.symfony.com/doc/3.x/tags/if.html
```
{{ YFParser('$(record : subject)$')|upper }}
{% if YFParser('$(record : subject)$') == 'record subject' %}
    ok
{% endif %}
```
https://twig.symfony.com/doc/3.x/tags/for.html
```
{% for recordData in YFParser('$(custom : RelatedRecordsData|name,assigned_user_id)$') %}
      name: {{ recordData.name }}<br>
      user: {{ recordData.assigned_user_id }}<br>
      {% if loop.last == false %}
            -------- <br>
      {% endif %}
{% endfor %}
```
> 4. I'd like to create plugins for my customers. Is there a way to install them via a zip file?

yes, you can install zip packages, you can create a module from the panel, you can export it and install it on another CRM
![image](https://user-images.githubusercontent.com/10198654/182716568-77cd3d9d-ab27-4cb4-8974-d9123f0d82f0.png)
![image](https://user-images.githubusercontent.com/10198654/182716514-b5f1facb-755d-4378-a601-678644e25880.png)

**Reply by bpabiszczak** [reply]
> I wouldn't like my clients to see this:
> 
> ![image](https://user-images.githubusercontent.com/36896420/182420405-c47345d9-2232-410e-80a2-157d1cf88aeb.png)
> 
> and this:
> 
> ![image](https://user-images.githubusercontent.com/36896420/182420501-3b4ef31b-2692-423d-a245-98ce0a816c26.png)

What if we showed your services and products here ?? : P

**Reply by rulotec** [reply]
That would be great. Could you elaborate on how that would work?

Regarding my questions 1, 3 and 5, could you please also comment on that.

Thanks in advance!

**Comment by rulotec** [comment]
I liked the fact that you have the manually triggered workflows. Also there's many field types you can create with the field editor.

However, I couldn't find a way to "convert" leads into accounts. I don't know if you have that concept in Yeti.

**Reply by rskrzypczak** [reply]
![image](https://user-images.githubusercontent.com/10221999/182520246-069ce968-f33e-4230-a638-5e2bde7e467d.png)
Configuration: _Software configuration/Processes/Marketing_
Mapping: _Software configuration/Standard modules/Mapping_

**Reply by mariuszkrzaczkowski** [reply]
there is another way to convert / generate records, see mapping mechanism:
https://gitdeveloper.yetiforce.com/index.php?module=MappedFields&parent=Settings&view=List
![image](https://user-images.githubusercontent.com/10198654/182536984-15262524-26eb-453d-9109-9f1c326abdd2.png)
https://gitdeveloper.yetiforce.com/index.php?module=SSingleOrders&view=List
![image](https://user-images.githubusercontent.com/10198654/182537032-7b3be711-f164-4193-b23d-bb18285fdf0c.png)

**Reply by rulotec** [reply]
Wow, the custom mapping mechanism is cool. Seems like you can create custom mappings between any two modules. That's pretty flexible!

In the case of the leads module, I see a Lead is more of a Company. In Vtiger, the Leads module is a combination of 3 things: a Company, a contact and an Opprotunity (i.e. you have fealds for each of those). Then when you convert, you "fork" the fields into the corresponding 3 modules: Account, Contact and Opprotunity.

So, in the case of Yeti, in a B2C Company where you only deal with people (say a university for example), how would you introduce the "Leads" into my system? Would I have to add fields like "Name" and "Lastname" to the current Leads module  and do a custom mapping?

Or would I have to create my own "Lead contacts" custom module to achieve that?

**Reply by mariuszkrzaczkowski** [reply]
> So, in the case of Yeti, in a B2C Company where you only deal with people (say a university for example), how would you introduce the "Leads" into my system? Would I have to add fields like "Name" and "Lastname" to the current Leads module and do a custom mapping?

look at this
![image](https://user-images.githubusercontent.com/10198654/182717226-87754e22-785d-4c31-979a-bf81352f5209.png)
![image](https://user-images.githubusercontent.com/10198654/182717241-bbb09310-fa78-483e-b898-ddebae72451f.png)

**Reply by rulotec** [reply]
Thanks @mariuszkrzaczkowski that clears it up.
Please guys don't forget to comment on my observetions regarding points 1, 3 and 5 of the original post by @bpabiszczak .

**Comment by angelopaglialonga81** [comment]
Hi, @mariuszkrzaczkowski I have a bunch of technical/functional questions for you:

1. how can I use custom vars or functions in email templates? I see there is a category under general vars.
2. How can I export a table "report" or widget in xls or send it to a manager every week?
3.  Is it possible to create a pivot report?
4. is it possible to create a custom sql report?
5. May I add custom expression functions in workflows?
6. May I create custom tasks in workflows just like in vtiger?
7. If I want to create a module for a client with a settings panel, may I reuse one already written in Yetiforce and start from there or do I have to rewrite it from scratch?
8. Is it possible to write drip emails using workflows?
9. May I mass email customers without Mailchimp or other integrations?

**Reply by mariuszkrzaczkowski** [reply]
> 1. how can I use custom vars or functions in email templates? I see there is a category under general vars.

specify the question
> 6. May I create custom tasks in workflows just like in vtiger?

yes, it works almost the same
> 8. Is it possible to write drip emails using workflows?

give an example
> 9. May I mass email customers without Mailchimp or other integrations?

yes, there is a mail template mechanism,
which adds messages to the queue and sends it in batches
![image](https://user-images.githubusercontent.com/10198654/182718011-89fa2451-2b3f-45a3-88b2-6673fae847c8.png)
![image](https://user-images.githubusercontent.com/10198654/182718034-fb1b607a-49ca-4e48-a248-72021e5d0b5f.png)

**Reply by angelopaglialonga81** [reply]
Good morning @mariuszkrzaczkowski thank you for helping by answering my questions.
About the questions that needed clarification:
1. How can I define custom vars in the menu "additional variables" of email templates?
![Schermata 2022-08-04 alle 09 32 39](https://user-images.githubusercontent.com/106532913/182790116-70e31cd0-2801-4f6a-a72b-3d7762e938c0.png)

8. drip email sequence:  send a mail to ask subscription confirmation, if user clicks "confirm" in email, then crm sends one mail after 5 minutes, the second mail after 1 day and the last mail after 3 days from subscription, is it possible? 

What about questions 2 to 5 and question 7?

**Reply by bpabiszczak** [reply]
@angelopaglialonga81 
1. you must add file to folder.
2. you can configure it via workflow, but it may require some extra script [maybe].
3. use standard record list view in module [you have tools such as export available there] [2]
4. yes, you add a new method with a script [as in vtiger] [5]

**Reply by angelopaglialonga81** [reply]
Thank you!

**Comment by s2b-git** [comment]
I will drop my coin:

The first issue you mentioned were some really good addons from Its4You (EmailMaker & PDFMaker)
Unfortunately nobody described exactly what’s missing in our built-in mechanisms by default and I can’t expect Its4You to do that since there’s nothing in it for them… I assume these addons contribute greatly to their income, meanwhile most of these functionalities are free and built-in in YetiForce.

**Certainly, with effort EmailMaker and PDFMaker can work nearly similarly in Yetiforce, the difference (especially regarding PDFMaker) would be user friendly design operations. Without much hassle in PDFMaker we can create descent and complex templates, which the same in Yetiforce will take a lot of more time plus without programming skills it will be impossible. 
It is hard to describe it properly, you will need to install one and test. Obviously if PDF Templater can be reengineered in the way PDFMaker is done and sold in your Marketplace I will be the one who will pay for that (and I am sure a lot of people will do the same).
The important part is WYSWIG. This is missing here**

Workflow Designer from Redoo networks
Similarly to the first question, I assume users who don’t know YetiForce well don’t know what automation mechanisms are built into the system. This extension has a great interface but I doubt it offers even part of the possibilities YetiForce offers by default.
Maybe someone could make a comparison, but IMO it’s fear of the unknown :)

**Workflow Designer from Redoo is something what cannot be compared I believe. It is very large addon with enormous number of predesigned elements, which partially can be used without programming skills, but most of it needs someone who really have a deeper clue. It is not only workflows, there is hell more bits which can extend vtiger  to completely new level of what vtiger is, so I will not even try to get it compared. Yeti have nicely done workflow part, not perfect here and there but doesn't cost 600 Euro and evolving for sure.** 

Cheers!

**Reply by garryatbrighton** [reply]
> This extension has a great interface but I doubt it offers even part of the possibilities YetiForce offers by default.

It offers much more... you are going for very hash assumption.

**Comment by rulotec** [comment, marked as answer]
There is also a well known extension called "Dynamic Fields" (https://it-solutions4you.com/vtiger-extensions/dynamic-fields/) which lets you modify fields visibility, mandatory status, etc, based on the user's actions. 

An example would be: You change a ticket status to "closed", and then the "solution description" field should become mandatory. It should not be mandatory beforehand because, of course,  you don't have a solution in the first stages of the ticket.

Is there something similar in Yeti?

**Reply by apcloic** [reply]
@rulotec 
Yes, you can check yetiforce dynamic fields built-in system here and compare with its4you solution, I think it's quite similar. 
https://gitstable.yetiforce.com/index.php?parent=Settings&module=FieldsDependency&view=List

**Reply by mariuszkrzaczkowski** [reply]
See also the extensive functionality "Fields - Picklists / Relations" on the developer version. You can easily dynamically check the values in all dictionaries
https://gitdeveloper.yetiforce.com/index.php?parent=Settings&module=PickListDependency&view=Edit
![image](https://user-images.githubusercontent.com/10198654/183018488-8f3fe503-b4f5-46b9-8f85-7b3e4493c233.png)

**Comment by rulotec** [comment]
Hi guys, 

I have some more questions.

1. I didn't find any webforms module. I guess I would have to create my own script that can handle post requests form a webform? or use the api?

2. I see there are many more field types in yeti as compared to vtiger, which is already great. Something I didn't like about vtiger is that if you want to add a new field type, you needed to edit vtiger core files, violating the "Open Closed Principle" of OOD. In Yeti, can I create a new field type without having to edit existing code (i.e. just adding a new file here and there)?

3. what about tabular custom reports? Reports where I can choose the columns and the conditions and have a scheduled sending via email.

4. Are there companies that build modules for yetiforce other than yetiforce itself? I couldn't find any after a quick google search.

**Reply by mariuszkrzaczkowski** [reply]
> 1. I didn't find any webforms module. I guess I would have to create my own script that can handle post requests form a webform? or use the api?

look for info in issues, this has been mentioned many times. A few quick links to look for:
https://github.com/YetiForceCompany/YetiForceScripts/tree/master/YetiREST
https://github.com/YetiForceCompany/YetiForceCRM/issues/10462
https://github.com/YetiForceCompany/YetiForceCRM/issues/7137
https://github.com/YetiForceCompany/YetiForceCRM/issues/13740
https://github.com/YetiForceCompany/YetiForceCRM/issues/4980

>  In Yeti, can I create a new field type without having to edit existing code (i.e. just adding a new file here and there)?
yes, check this out:
https://github.com/YetiForceCompany/YetiForceCRM/commit/8d3b5aab9dc34d0bebbad5bf240c1e1a1bd804bd
https://github.com/YetiForceCompany/YetiForceCRM/commit/27c8ca343f3ef9dbcb1a9c60e0e940f56839e873
https://github.com/YetiForceCompany/YetiForceCRM/commit/40bcb2ac748c31ac6317da7f6a2a712035fc2c3d

**Reply by apcloic** [reply]
@rulotec Also you may check this one for Google forms to Yetiforce.
#13933

**Reply by rulotec** [reply]
@mariuszkrzaczkowski I see you still have to edit core files in order to add a custom field type (e.g. modules/Settings/LayoutEditor/models/Module.php and modules/Vtiger/models/Field.php). It's way better than vtiger, where you have if and switch statements everywhere regarding uitypes. The perfect solution in my opinion would be to store every configuration aspect of the field type in the DB, like for example, this:

![image](https://user-images.githubusercontent.com/36896420/183319077-60061270-154a-493a-90d5-cb2ee2893a2c.png)

so you can truly add a new field type without editing a single core file (just like when adding a custom module). But from what I could quickly see in the code, Yeti is waaaay closer to that than Vtiger.

**Reply by mariuszkrzaczkowski** [reply]
>  I see you still have to edit core files in order to add a custom field type (e.g. modules/Settings/LayoutEditor/models/Module.php and modules/Vtiger/models/Field.php). It's way better than vtiger, where you have if and switch statements everywhere regarding uitypes. 

modules/Settings/LayoutEditor/models/Module.php - The file is not required to add a field, it only adds a field to the interface (field creation window)
modules/Vtiger/models/Field.php - yes, we only add system fields, there is an alternative, you just need to add an entry to the database (`vtiger_ws_fieldtype`) and you do not need to edit the file
![image](https://user-images.githubusercontent.com/10198654/183351989-a0f45a1f-b716-4860-8430-8a97f6d8af40.png)

>  The perfect solution in my opinion would be to store every configuration aspect of the field type in the DB, like for example, this:
> so you can truly add a new field type without editing a single core file (just like when adding a custom module). But from what I could quickly see in the code, Yeti is waaaay closer to that than Vtiger.

we are planning it, only the entire reconstruction and development take a lot of time

**Comment by Marshal4582** [comment]
My wish-list :o)

1. Navigation in details view is missing. Also "Save and Next" in full edit record view will nicely improve speed of work. Those are crucial for big data sets
2. Double click to edit record in list view (mentioned in some discussions)
3. Configurable shortcuts (current shortcuts are in conflict with various browsers default shortcut settings). Allowing to move right/left with shortcuts in details view
4. Conversion conditions in mailscanner (will improve tickets management)
